music.py

```diff
@@ -20,16 +22,118 @@ def play_drumbase(beats, duration):
 
 
 def tone(n, base_freq=440.0):
-    #  0 A
-    #  1 A#
-    #  2 B
-    #  3 C
-    #  4 C#
-    #  5 D
-    #  6 D#
-    #  7 E
-    #  8 F
-    #  9 F#
-    # 10 G
-    # 11 G#
+    """Return the frequency of the nth interval from base_freq (in 12-TET)."""
+    # -2 -1 0  1  2  3  4  5  6  7  8  9  10 11 12
+    # G  G# A  A# B  C  C# D  D# E  F  F# G  G# A
+    # G  Ab A  Bb B  C  Db D  Eb E  F  Gb G  Ab A
     return base_freq * 2 ** (n/12)
+
+
+# This dict maps the name of the notes from C0 (included) to C8 (excluded)
+# to the corresponding frequency (in 12-TET).
+tones = [tone(i) for i in range(-57, 39)]
+names_sharp = ['C', 'C#', 'D', 'D#', 'E', 'F', 'F#', 'G', 'G#', 'A', 'A#', 'B']
+names_flat  = ['C', 'Db', 'D', 'Eb', 'E', 'F', 'Gb', 'G', 'Ab', 'A', 'Bb', 'B']
+notes = {}
+
+for t, ns, nf, o in zip(tones, cycle(names_sharp), cycle(names_flat),
+                        chain.from_iterable(repeat(o, 12) for o in range(8))):
```

Comment:
The chained iterator looks convoluted, I suggest to expand it as a generator function, like:
```suggestion
def octaves():
    for o in range(8):
        for _ in range(12):
            yield o

for t, ns, nf, o in zip(tones, cycle(names_sharp), cycle(names_flat),
                        octaves()):
```

Reply:
I refactored it a bit to make it clearer.